scripts/zones/Bastok_Markets/npcs/Matthias.lua

```diff
@@ -15,17 +15,125 @@ require("scripts/zones/Bastok_Markets/TextIDs");
+                trade:getGil() == 0) then
+				
+                rewardThePlayer(player);
+				
+             elseif(playersAFChoice == 3 and --wamoura cloth: 2289, moblinweave: 1636, rainbow velvet: 1999
+                trade:hasItemQty(2289, 1) == true and
+                trade:hasItemQty(1636, 1) == true and
+                trade:hasItemQty(1999, 1) == true and
+                trade:getItemCount() == 3 and
+                trade:getGil() == 0) then
+				
+                rewardThePlayer(player);
+            end;
+     end;
 end; 
 
+function rewardThePlayer(player)
+    local playersAFChoice = player:getVar("dancerAFChoice");
+    local currentVanaDay = VanadielDayOfTheYear();
+    player:setVar("dancerTailorWorkDay", currentVanaDay);
+    player:setVar("dancerTailorCS", 5);
+    player:tradeComplete();
+    player:startEvent(0x01EF, playersAFChoice-1);
+end;
 
 -----------------------------------
 -- onTrigger Action
 -----------------------------------
 
 function onTrigger(player,npc)
-
-    player:startEvent(0x01f3);
-
+	
+    local playersAFChoice = player:getVar("dancerAFChoice");
+    local tailorStartedWorkDay = player:getVar("dancerTailorWorkDay");
+	
+    if(player:getVar("dancerTailorCS") == 2) then
+        player:startEvent(0x01EC);
+    elseif(player:getVar("dancerTailorCS") == 3) then
+		
+        local completedPieces = player:getVar("dancerCompletedAF");
+		
+        local playerCompletedTiara = 0;
+        if(bit.band(completedPieces,1) > 0) then
+            playerCompletedTiara = 1;
+        end;
+		
+        local playerCompletedBangles = 0;
+        if(bit.band(completedPieces,2) > 0) then
+            playerCompletedBangles = 1;
+        end;
+		
+        local playerCompletedShoes = 0;
+        if(bit.band(completedPieces,4) > 0) then
+            playerCompletedShoes = 1;
+        end;
+		
+        local completedPieces = playerCompletedShoes + playerCompletedBangles + playerCompletedTiara;
+        if(completedPieces == 3) then 
+            player:setVar("dancerTailorCS", 6);
+            player:startEvent(0x01F2);
+         else
+            player:startEvent(0x01ED, playerCompletedTiara, playerCompletedBangles, playerCompletedShoes);
+        end;
+		
+    elseif(player:getVar("dancerTailorCS") == 4) then
+			
+        player:startEvent(0x01EE, playersAFChoice -1); -- event params indexed from 0
+	
+    elseif(player:getVar("dancerTailorCS") == 5 )then
+	
+        local currentVanaDay = VanadielDayOfTheYear();
+		
+        if(player:getFreeSlotsCount() == 0) then
+		
+            player:PrintToPlayer( "Not enough free inventory space." );
+			
+        elseif(currentVanaDay > tailorStartedWorkDay ) then
+			
+            local dancerAFID = 1; -- variable used to convert player's choice into an Item ID. gender is actually important here because it displays the item on screen for you
+            local playerGender = player:getGender();
+			
+            switch (playersAFChoice) : caseof{
```

Comment:
stick a space between ')' and 'then', and remove any extra ' ' before the closing ')'
also, stick a space between if and parentheses so it looks like this
```
if (player:getFreeSlotsCount() == 0) then
```